### PR TITLE
Search: Do not redirect to the same page in wp-admin

### PR DESCRIPTION
**Pull request by kraftbj**
Fixes pbfA90-tz-p2

#### Changes proposed in this Pull Request:

* Do not redirect when deactivating Search.

#### Jetpack product discussion

Debug notes: p1648143618433359-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Before this patch, spin up a JN running JP master. Don't need to connect even, but activate Jetpack and attempt to go to Appereance->Menus.
* Expected: It goes to the nav-menu.php page.
* Before this patch, it reloads the page you're on.

### CODE REVIEW

File: projects/packages/search/src/class-module-control.php [hunk 1]
```diff
@@ -123,7 +123,7 @@ public function deactivate() {
 		 *
 		 * @param string $module Module slug.
```
Comment (anomiex) [inline projects/packages/search/src/class-module-control.php:124]: Document the new parameter?

**Review by anomiex — COMMENTED** [review]
Should the call at
https://github.com/Automattic/jetpack/blob/a9afe371039575add9459ed6fa13d21ff522bdbf/projects/plugins/jetpack/class.jetpack.php#L3064-L3071
be updated too?

**Comment by kraftbj** [comment]
> Should the call at .... be updated too?

That's a good question. My intent with this PR is to correct the breakage that led to today's release being reverted on Atomic, which is related to the Search code background deactivating modules. I didn't research why that redirect code is present at all, but it's old ( https://github.com/Automattic/jetpack/pull/339 )

**Comment by anomiex** [comment]
Note by "updated" I just meant documenting the new parameter and passing true for it to match the current implicit default.

**Comment by kraftbj** [comment]
@anomiex I'm going to update it, but I don't like what's going on there. The same function is hooked onto two filters with different args for each filter. We need to clean it up, but rather do that in a separate commit.

**Review by samiff — APPROVED** [review]
Works for me 👍  

Merged master to unstuck the check reporting (jobs were done 🤷 )

**Comment by samiff** [comment]
picked to jetpack/branch-10.8 in 63f81210dbc82047f41a2b325a929480759ae522
packages/search v0.11.3 released in #23611

**Pull request merged**